providers/easyname/provider.go

```diff
@@ -0,0 +1,70 @@
+package easyname
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+)
+
+type easynameProvider struct {
+	apikey   string
+	apiauth  string
+	signSalt string
+	domains  map[string]easynameDomain
+}
+
+func init() {
+	providers.RegisterRegistrarType("EASYNAME", newEasyname)
+}
+
+func newEasyname(m map[string]string) (providers.Registrar, error) {
+	api := &easynameProvider{}
+
+	if m["email"] == "" || m["userid"] == "" || m["apikey"] == "" || m["authsalt"] == "" || m["signsalt"] == "" {
+		return nil, fmt.Errorf("missing easyname email, userid, apikey, and/or salt")
+	}
+
+	api.apikey, api.signSalt = m["apikey"], m["signsalt"]
+	composed := fmt.Sprintf(m["authsalt"], m["userid"], m["email"])
```

Comment:
Missing format string.

Reply:
No, `authsalt` is the format string ;-)  It's a really weird system.